crates/accelerate/src/circuit_library/multi_local.rs

```diff
@@ -0,0 +1,317 @@
+///     3-tuple elements containing (packed_operation, num_qubits, num_params).
+/// - `parameters`: The set of parameter objects to use for the operations. This is a 3x nested
+///     vector, organized as operation -> block -> param. That means that for operation `i`
+///     and block `j`, the parameters are given by `parameters[i][j]`.
+///
+/// # Returns
+///
+/// An iterator for the entanglement instructions.
+fn entanglement_layer<'a>(
+    py: Python<'a>,
+    entanglement: &'a LayerEntanglement,
+    entanglement_blocks: &'a [&'a Block],
+    parameters: LayerParameters<'a>,
+) -> impl Iterator<Item = PyResult<Instruction>> + 'a {
+    let zipped = izip!(entanglement_blocks, parameters, entanglement);
+    zipped.flat_map(move |(block, block_params, block_entanglement)| {
+        block_entanglement
+            .iter()
+            .zip(block_params)
+            .map(move |(indices, params)| {
+                let (bound_op, bound_params) = block
+                    .operation
+                    .assign_parameters(py, &params)
+                    .expect("Failed to rebind");
+                Ok((
+                    bound_op,
+                    bound_params,
+                    indices.iter().map(|i| Qubit(*i)).collect(),
+                    vec![] as Vec<Clbit>,
+                ))
+            })
+    })
+}
+
+/// # Arguments
+///
+/// - `num_qubits`: The number of qubits of the circuit.
+/// - `rotation_blocks`: The blocks used in the rotation layers. If multiple are passed,
+///     these will be applied one after another (like new sub-layers).
+/// - `entanglement_blocks`: The blocks used in the entanglement layers. If multiple are passed,
+///     these will be applied one after another.
+/// - `entanglement`: The indices specifying on which qubits the input blocks act. This is
+///     specified by string describing an entanglement strategy (see the additional info)
+///     or a list of qubit connections.
+///     If a list of entanglement blocks is passed, different entanglement for each block can
+///     be specified by passing a list of entanglements. To specify varying entanglement for
+///     each repetition, pass a callable that takes as input the layer and returns the
+///     entanglement for that layer.
+///     Defaults to ``"full"``, meaning an all-to-all entanglement structure.
+/// - `reps`: Specifies how often the rotation blocks and entanglement blocks are repeated.
+/// - `insert_barriers`: If ``True``, barriers are inserted in between each layer. If ``False``,
+///     no barriers are inserted.
+/// - `parameter_prefix`: The prefix used if default parameters are generated.
+/// - `skip_final_rotation_layer`: Whether a final rotation layer is added to the circuit.
+/// - `skip_unentangled_qubits`: If ``True``, the rotation gates act only on qubits that
+///     are entangled. If ``False``, the rotation gates act on all qubits.
+///
```

Comment:
what about the `overwrite_block_parameters, inital_state, name, flatten` parameters?

Reply:
These don't have analogs in the new function. The new function doesn't expose them because they aren't really necessary in the new form of returning a bare `QuantumCircuit` directly. If you want to something like prepend an initial state you can use `compose()` or the other circuit apis for that functionality.